plugins/de.cognicrypt.staticanalyzer.kotlin/src/de/cognicrypt/staticanalyzer/kotlin/utilities/KotlinUtils.java

```diff
@@ -0,0 +1,261 @@
+ 						if(result.compiledCorrectly())
+							Activator.getDefault().logInfo("Finished compiling kotlin files.");
+						else
+							Activator.getDefault().logInfo("Cannot compile some kotlin files due to errors. Static analysis skipped them.");
+ 					}	
+ 				} catch (CoreException | OperationCanceledException | InterruptedException e) {
+ 					e.printStackTrace();
+ 				}
+ 			}
+ 		}
+	}
+	
+	public static IResource findKotlinClassByName(final String className, final IProject currentProject)
+			throws JavaModelException, ClassNotFoundException {
+		if(Platform.getBundle("org.jetbrains.kotlin.core") != null) {
+			// This part is required because Eclipse JDT doesn’t provide any mapping of kotlin light classes to its source code
+			// As a result the above IPackageFragment.getCompilationUnits() doesn't return any kotlin .class files
+			if(KotlinNature.hasKotlinNature(currentProject)) {
+
+				// computing corresponding source filename, since in kotlin .class filename is changed
+				// Eg. Demo.kt is compiled to DemoKt.class
+				String[] temp = className.split("\\.");
+				String classFileName = temp[temp.length-1];
+				String srcFilename = "";
+
+				if(classFileName.substring(classFileName.length()-2).equals("Kt")) {
+					srcFilename = classFileName.substring(0, classFileName.length()-2) + ".kt";
+				}
+				else if(classFileName.contains("$")) {
+					srcFilename = Arrays.asList(classFileName.split("\\$")).get(0) + ".kt";
+				}
+				// because in some projects the class names aren't renamed
+				else
+					srcFilename = classFileName + ".kt";
+
+				for (final IPackageFragment l : JavaCore.create(currentProject).getPackageFragments()) {
+					// this check is needed because IJavaProject.getPackageFragments() returns dependencies as well
+					if(l.getKind() == IPackageFragmentRoot.K_SOURCE) {
+						// removing the <project_name> from path returned by IPackageFragment.getPath() because IProject.getFile() also appends it
+						String[] originalPath = l.getPath().toString().split(File.separator);
+						String[] modifiedPath = Arrays.copyOfRange(originalPath, 2, originalPath.length);
+						String packageName = String.join(File.separator, modifiedPath);
+
+						IFile sourceFile = currentProject.getFile(packageName + File.separator + srcFilename);
+						if(sourceFile.exists())
+							return (IResource) sourceFile;
+					}
+				}
+			}
+		}
+		
+		throw new ClassNotFoundException("Class " + className + " not found.");
+	}
+	
+	public static boolean verifyKotlinDependency(IJavaProject javaProject) {
+ 		IProject p = javaProject.getProject();
+ 		IFile pomFile = p.getFile("pom.xml");
+ 		boolean isPresent = false;
+ 		try {
+ 			MavenProject project = loadProject(pomFile.getLocation().toFile());
+ 			List<Dependency> dependencies = project.getDependencies();
+ 			
+ 			for (Dependency dependency : dependencies) {
+ 				System.out.println(dependency);
+ 				if(dependency.getArtifactId().equals("kotlin-stdlib"))
```

Comment:
Change to "kotlin-stdlib".equals(dependency.getArtifactId(). Avoids some potential NPEs.